x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/stats/Metrics.java

```diff
@@ -56,6 +59,9 @@ public Metrics() {
             fMap.put(featureMetric, new CounterMetric());
         }
         featuresMetrics = Collections.unmodifiableMap(fMap);
+
+        // we could pre-populate this map using FunctionRegistry
+        functionMetrics = Collections.synchronizedMap(new HashMap<>());
```

Comment:
Pre-populating would indeed make this more in line with the rest of how the usage is reported, no?

Otherwise, there are never `0` usages of functions reported, just values larger than zero or the function is missing altogether. Won't it be beneficial for e.g. making visualizations when each function is actually reported?

Reply:
My concern is that it would return a huge list (potentially hundreds of entries), but probably it still makes sense in terms of consistency

Reply:
Well, after the cluster's been up for a while, the list of used functions may be near exhaustive anyway, no?

If the number of functions returned is a concern, is it safe to report their usage in the first place?

Maybe better if e.g. @astefan weighed in here as I'm not familiar enough with how this API is consumed.